plugins/chat/assets/javascripts/discourse/initializers/chat-sidebar.js

```diff
@@ -471,18 +507,23 @@ export default {
               }
 
               get suffixValue() {
-                return this.channel.tracking.unreadCount > 0 ||
-                  this.channel.unreadThreadsCountSinceLastViewed > 0
-                  ? "circle"
-                  : "";
+                if (
+                  this.siteSettings.star_chat_channels &&
+                  this.channel.currentUserMembership.starred
+                ) {
```

Comment:
Same here, I think in this case you can just check `currentUserMembership.starred` and not the site setting, since `starred` will never be true if the site setting is false because of your serializer